docs/docs-beta/docs/guides/external-systems/non-python.md

```diff
@@ -1,4 +1,141 @@
 ---
 title: "Running non-Python languages"
 sidebar_position: 60
----
\ No newline at end of file
+---
+
+Dagster is written in Python, but that doesn't mean it's limited to running Python to materialize assets. This guide covers how to run some JavaScript with Dagster using Pipes.
+
+# Define your JavaScript function
+
+Here's a contrived example of some JavaScript code to run using Dagster Pipes. `train_model` load some dataset and train a sequential model using tensorflow.
+
+```javascript
+import * as tf from '@tensorflow/tfjs';
+
+async function train_model(config) {
+  const { path_to_data, data_config, path_to_model } = config;
+  const dataset = await tf.data.csv(path_to_data, data_config).map(({ xs, ys }) => {
+    return {
+      xs: tf.tensor2d(Object.values(xs), [Object.values(xs).length, 1]),
+      ys: tf.tensor2d(Object.values(ys), [Object.values(ys).length, 1])
+    };
+  })
+  .batch(100);
+
+  const model = tf.sequential()
+  model.add(tf.layers.dense({units: 1, inputShape: [1]}));
+  model.compile({loss: 'meanSquaredError', optimizer: 'sgd'});
+
+  await model.fitDataset(dataset, {epochs: 250})
+  await model.save(path_to_model);
+  model.summary();
+  return model;
+}
+```
+
+# Define your asset in Dagster
+
+<CodeExample filePath="guides/automation/pipes-asset.py" language="python" title="Asset using Dagster Pipes." />
+
+Dagster Pipes follows a similar design to Unix pipes, hence the name. The `PipesSubprocessClient` is responsible for running external processes and setting up input/output files. The asset defined here is materialized using the `PipesSubprocessClient` running a Node.js file containing the `train_model` function.
+
+# Add JavaScript utility functions to access pipes
```

Comment:
Possible title change, "Accessing Dagster context from our JavaScript code"

The idea of "accessing pipes" may be unclear to those who are new to it. Perhaps we change the title to the above, and briefly mention that this can be used to retrieve data from Dagster, and log events and metadata back.